nautobot/extras/models/name_color_content_types.py

```diff
@@ -0,0 +1,88 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.urls import reverse
+
+from nautobot.core.fields import AutoSlugField
+from nautobot.core.models import BaseModel
+from nautobot.extras.models.customfields import CustomFieldModel
+from nautobot.extras.models.change_logging import ChangeLoggedModel
+from nautobot.extras.models.relationships import RelationshipModel
+from nautobot.extras.models.mixins import DynamicGroupMixin, NotesMixin
+from nautobot.utilities.choices import ColorChoices
+from nautobot.utilities.fields import ColorField
+from nautobot.utilities.querysets import RestrictedQuerySet
+
+
+class ContentTypeRelatedQuerySet(RestrictedQuerySet):
+    def get_for_model(self, model):
+        """
+        Return all `self.model` instances assigned to the given model.
+        """
+        content_type = ContentType.objects.get_for_model(model._meta.concrete_model)
+        return self.filter(content_types=content_type)
+
+    def get_by_natural_key(self, name):
+        return self.get(name=name)
+
+
+# Inheriting from OrganizationalModel here causes partial import error
+class NameColorContentTypesModel(
+    BaseModel,
+    ChangeLoggedModel,
+    CustomFieldModel,
+    RelationshipModel,
+    NotesMixin,
+    DynamicGroupMixin,
+):
+    """
+    This abstract base properties model contains fields and functionality that are
+    shared amongst models that requires these fields: name, color, content_types and description.
+    """
+
+    # Todo(timizuo): Tag should inherit from this model; but
+    #  cant because of field conflicts: name and slug field.
```

Comment:
`TODO` what do you propose here?

Reply:
I'm not intending on fixing this in this PR or any other PR relating to this issue

Reply:
What do you propose we do for `Tag` then?

Reply:
replace `TagBase` with `NameColorContentTypesModel`